model_analyzer/monitor/record/memory.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from . import Record
+
+
+class MemoryRecord(Record):
```

Comment:
Filename should match class name, `memory_record.py`

Reply:
I would also like to point out that this may be a prime opportunity for a visitor pattern (https://refactoring.guru/design-patterns/visitor/python/example). Essentially, you would have a `record` base type that can be visited, then you have many different record types (MemoryRecord, ThroughputRecord, etc.). Then if there is an array of all records, a visitor, such as a report writer, can visit each record and dump the contents.

Not saying we will do that right now, but you should just be aware of the possibility.

Reply:
Is see. We can either do that or add a writer to the `RecordCollector`. I think @xprotobeast2 is working on the writers.